module/VuFind/tests/unit-tests/src/VuFindTest/I18n/SorterTest.php

```diff
@@ -0,0 +1,353 @@
+                    'locale' => 'en',
+                    'respectLocale' => false,
+                ],
+                -1
+            ],
+            [
+                [
+                    'strings' => ['a', 'a'],
+                    'locale' => 'en',
+                    'respectLocale' => false,
+                ],
+                0
+            ],
+            [
+                [
+                    'strings' => ['b', 'a'],
+                    'locale' => 'en',
+                    'respectLocale' => false,
+                ],
+                1
+            ],
+            [
+                [
+                    'strings' => ['a', 'A'],
+                    'locale' => 'en',
+                    'respectLocale' => false,
+                ],
+                0
+            ],
+            [
+                [
+                    'strings' => ['a', 'b'],
+                    'locale' => 'en',
+                    'respectLocale' => true,
+                ],
+                -1
+            ],
+            [
+                [
+                    'strings' => ['a', 'a'],
+                    'locale' => 'en',
+                    'respectLocale' => true,
+                ],
+                0
+            ],
+            [
+                [
+                    'strings' => ['b', 'a'],
+                    'locale' => 'en',
+                    'respectLocale' => true,
+                ],
+                1
+            ],
+            [
+                [
+                    'strings' => ['a', 'A'],
+                    'locale' => 'en',
+                    'respectLocale' => true,
+                ],
+                -1
+            ],
+            [
+                [
+                    'strings' => ['č', 'd'],
+                    'locale' => 'cs',
+                    'respectLocale' => false,
+                ],
+                1
+            ],
+            [
+                [
+                    'strings' => ['č', 'd'],
+                    'locale' => 'cs',
+                    'respectLocale' => true,
+                ],
+                -1
+            ],
+        ];
+    }
+
+    /**
+     * Test compare function
+     *
+     * @dataProvider compareProvider
+     *
+     * @return void
+     */
+    public function testCompare($test, $expected)
+    {
+        $sorter = $this->createSorter($test['locale'], $test['respectLocale']);
+        $result =  $sorter->compare($test['strings'][0], $test['strings'][1]);
+        if ($expected === 1) {
+            $this->assertGreaterThanOrEqual($expected, $result);
+        } elseif ($expected === -1) {
+            $this->assertLessThanOrEqual($expected, $result);
+        } else {
+            $this->assertEquals($expected, $result);
+        }
+    }
+
+    /**
+     * Data Provider for sort tests
+     *
+     * @return array
+     */
+    public static function sortProvider(): array
+    {
+        return [
+            [
```

Comment:
I am aware of ability to name test using array key, but it was enough for me the default behaviour which prints the whole data item in case the test fails ;)